lib/compat/wordpress-6.2/class-wp-theme-json-resolver-6-2.php

```diff
@@ -32,4 +32,27 @@ public static function theme_has_support() {
 		return wp_theme_has_theme_json();
 	}
 
+	/**
+	 * Builds the path to the given file and checks that it is readable.
+	 *
+	 * If it isn't, returns an empty string, otherwise returns the whole file path.
+	 *
+	 * @since 5.8.0
+	 * @since 5.9.0 Adapted to work with child themes, added the `$template` argument.
+	 * @deprecated 6.2.0 Use wp_get_theme()->get_file_path() instead.
+	 *
+	 * @param string $file_name Name of the file.
+	 * @param bool   $template  Optional. Use template theme directory. Default false.
+	 * @return string The whole file path or empty if the file doesn't exist.
+	 */
+	protected static function get_file_path_from_theme( $file_name, $template = false ) {
+		_deprecated_function( __METHOD__, '6.2.0', 'wp_get_theme()->get_file_path()' );
```

Comment:
These methods don't do the same, so I'm a bit wary about considering this a replacement.

The difference that I consider relevant enough here is that `wp_get_theme()->get_file_path()` does not have the `is_readable()` call, so it's not a direct replacement.

Maybe we can introduce a `wp_get_theme()->get_readable_file_path()` that includes that and returns an empty string if the file is not readable? With that I would agree that this can be deprecated.

Alternatively, if we want to not introduce this method and going forward keep the `is_readable()` calls separate, we should clarify here that you _also_ need to use `is_readable()`.

Reply:
As this method will not be used in core, I could just revert it and keep it as it is was. Friendly reminder, this is a private class and method, so extremely unlikely to be used.

Reply:
> this is a private class and method, so extremely unlikely to be used.

That's good insight, I didn't notice it. Would it make sense to remove this method entirely then?

Reply:
@mmtr Methods are not every removed in WordPress core. Only deprecated. Backwards compatibility.

Reply:
The [docstring of the `WP_Theme_JSON_Resolver` class](https://core.trac.wordpress.org/browser/trunk/src/wp-includes/class-wp-theme-json-resolver.php?rev=54824#L14) mentions that `This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes). This is a low-level API that may need to do breaking changes.`, so I thought this would be one of these times when it's ok to don't keep backwards compatibility.

Reply:
Said that, if we want to keep this method around, I think the deprecation warning should provide a more accurate replacement as @felixarntz suggested above. However I feel that something like the below is too much verbose, but cannot think of anything better 🤔.

```suggestion
	protected static function get_file_path_from_theme( $file_name, $template = false ) {
		_deprecated_function( __METHOD__, '6.2.0', 'is_readable( wp_get_theme()->get_file_path() ) ? wp_get_theme()->get_file_path() : \'\'' );
```

Reply:
Hmm, given the private _and_ internal angle I would agree that we can just remove the method. Not common in core, but this is different IMO.

Reply:
I have restored the method and added a todo here. Let remove this method in core, it is a pointless method anyway.